org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenProjectFacade.java

```diff
@@ -628,7 +654,7 @@ private MavenProject tryGetMavenProject() {
       return manager.getMavenProject(this, null);
     } catch(CoreException ex) {
       //TODO can we handle this more graceful? E.g. one error condition is that project dependencies can not be resolved, maybe we can try to load the project without dependencies then?
-      manager.getMarkerManager().addErrorMarkers(pom, IMavenConstants.MARKER_POM_LOADING_ID, ex);
+      logException(ex);
```

Comment:
same here, better throw the exception to the caller instead and handel it there

Reply:
Same as above. Called by `IMavenProjectFacade.createExecutionContext()`.

Reply:
One way would be to return an "ExceptionalExecutionContext" that throws the original exception on any call of its methods (that allow CoreException already), this is also done with the Plexuscontainer as well.